pkg/model/components/apiserver.go

```diff
@@ -115,6 +115,10 @@ func (b *KubeAPIServerOptionsBuilder) BuildOptions(o interface{}) error {
 		return fmt.Errorf("unknown cloudprovider %q", clusterSpec.CloudProvider)
 	}
 
+	if clusterSpec.ExternalCloudControllerManager != nil {
```

Comment:
Adding this here so we can discuss later. I don't think setting this flag is actually required for apiserver, leaving the cloudprovider blank is sufficient, same for KCM. No change required for this PR just a note for the future

Reply:
@wlan0 please file an issue and make a note in the code